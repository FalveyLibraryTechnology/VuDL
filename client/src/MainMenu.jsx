import React from "react";
import { Link } from "react-router-dom";

export default function MainMenu() {
    return (
        <div>
            <h1>VuDL Admin</h1>
            <h2>Core Functions</h2>
            <ul>
                <li>
                    <Link to="/paginate">Job Paginator</Link>
                </li>
                <li>Object Editor [coming soon]</li>
            </ul>
            <h2>Other Tools</h2>
            <ul>
                <li>
                    <Link to="/pdf">PDF Generator</Link>
                </li>
                <li>
                    <Link to="/solr">Solr Indexer</Link>
                </li>
            </ul>
        </div>
    );
}
