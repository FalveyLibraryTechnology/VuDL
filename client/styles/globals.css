:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
