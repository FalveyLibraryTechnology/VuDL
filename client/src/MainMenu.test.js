import React from "react";
import { describe, expect, it } from "@jest/globals";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MainMenu from "./MainMenu";

describe("MainMenu", () => {
    it("renders", () => {
        const wrapper = shallow(<MainMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
