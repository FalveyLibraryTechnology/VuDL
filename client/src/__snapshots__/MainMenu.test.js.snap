// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu renders 1`] = `
<div
  className="main-menu"
>
  <h1
    className="main-menu-heading"
  >
    VuDL Admin
  </h1>
  <h2
    className="main-menu-core-functions-heading"
  >
    Core Functions
  </h2>
  <ul
    className="main-menu-core-functions-list"
  >
    <li>
      <Link
        to="/paginate"
      >
        Job Paginator
      </Link>
    </li>
    <li>
      Object Editor [coming soon]
    </li>
  </ul>
  <h2
    className="main-menu-other-tools-heading"
  >
    Other Tools
  </h2>
  <ul
    className="main-menu-other-tools-list"
  >
    <li>
      <a
        href="http://localhost:9000/queue"
      >
        Arena (Job Queue Manager)
      </a>
    </li>
    <li>
      <Link
        to="/pdf"
      >
        PDF Generator
      </Link>
    </li>
    <li>
      <Link
        to="/solr"
      >
        Solr Indexer
      </Link>
    </li>
  </ul>
</div>
`;
